src/main/java/com/shopify/minishopify/controllers/ShopController.java

```diff
@@ -0,0 +1,63 @@
+package com.shopify.minishopify.controllers;
+
+import com.shopify.minishopify.model.Shop;
+import com.shopify.minishopify.repository.ShopRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+
+import java.util.List;
+import java.util.Optional;
+
+@RestController
+@RequestMapping("shops")
+public class ShopController {
+
+    @Autowired
+    private ShopRepository shopRepository;
+
+    @GetMapping(value="/{id}", produces=MediaType.APPLICATION_JSON_VALUE)
+    @ResponseBody
+    public Shop getShopById(@PathVariable int id) {
+
+        Optional<Shop> shopQuery = shopRepository.findById(id);
+
+        if (shopQuery.isPresent()) {
+
+            return shopQuery.get();
+        } else {
+            throw new ResponseStatusException(HttpStatus.NOT_FOUND, String.format("No shop found with id: %d", id));
+        }
+    }
+
+    @GetMapping("/")
+    public List<Shop> getAllShops() {
+
+        return shopRepository.findAll();
+    }
+
+    @PostMapping("/create")
+    public Shop createShop(@RequestBody Shop shop) {
```

Comment:
Are we expecting the client to create the Shop, then serialize it to JSON? Wouldn't it be better to consume the parameters, and create it on the server?

Ditto goes for "updateShop".

Reply:
I'm not sure what you mean by parameters; I'm going to assume you mean the form fields that are sent when a client submits a form.

I decided for JSON because its more flexible for different use cases. This is more true for `updateShop`. For example, if someone adds a new `Category` to a `Shop`, it doesn't really make sense for the entire shop form to be there so the `updateShop` has all the necessary parameters. 

For `createShop`, the above paragraph is less true but I think REST APIs generally consume JSON from clients. This could probably be changed.